guides/source/active_record_callbacks.md

```diff
@@ -329,124 +763,267 @@ Just as with validations, it is also possible to skip callbacks by using the fol
 
-Callbacks work through model relationships, and can even be defined by them. Suppose an example where a user has many articles. A user's articles should be destroyed if the user is destroyed. Let's add an `after_destroy` callback to the `User` model by way of its relationship to the `Article` model:
+Let's consider a scenario where we have a `User` model with a callback that
+sends a welcome email to new users after they sign up. However, there might be
+cases where we want to create a user without sending the welcome email, such as
+during seeding the database with test data.
 
 ```ruby
 class User < ApplicationRecord
-  has_many :articles, dependent: :destroy
+  after_create :send_welcome_email
+
+  def send_welcome_email
+    puts "Welcome email sent to #{self.email}"
+  end
 end
+```
 
-class Article < ApplicationRecord
-  after_destroy :log_destroy_action
+In this example, the `after_create` callback triggers the `send_welcome_email`
+method every time a new user is created.
 
-  def log_destroy_action
-    puts 'Article destroyed'
+To create a user without sending the welcome email, we can use the
+`ActiveRecord::Suppressor` module as follows:
+
+```ruby
+User.suppress do
+  User.create(name: "Jane", email: "jane@example.com")
+end
+```
+
+In the above code, the `User.suppress` block ensures that the
+`send_welcome_email` callback is not executed during the creation of the "Jane"
+user, allowing us to create the user without sending the welcome email.
+
+WARNING: Using the Active Record Suppressor, while potentially beneficial for
+selectively controlling callback execution, can introduce complexity and
+unexpected behavior. Suppressing callbacks can obscure the intended flow of your
+application, leading to difficulties in understanding and maintaining the
+codebase over time. Carefully consider the implications of suppressing
+callbacks, ensuring thorough documentation and thoughtful testing to mitigate
+risks of unintended side effects, performance issues, and test failures.
+
+Halting Execution
+-----------------
+
+As you start registering new callbacks for your models, they will be queued for
+execution. This queue will include all of your model's validations, the
+registered callbacks, and the database operation to be executed.
+
+The whole callback chain is wrapped in a transaction. If any callback raises an
+exception, the execution chain gets halted and a **rollback** is issued, and the
+error will be re-raised.
+
+```ruby
+class Product < ActiveRecord::Base
+  before_validation do
+    raise "Price can't be negative" if total_price < 0
   end
 end
+
+Product.create # raises "Price can't be negative"
 ```
 
-```irb
-irb> user = User.first
-=> #<User id: 1>
-irb> user.articles.create!
-=> #<Article id: 1, user_id: 1>
-irb> user.destroy
-Article destroyed
-=> #<User id: 1>
+This unexpectedly breaks code that does not expect methods like `create` and
+`save` to raise exceptions.
+
```

Comment:
Can  we move the warning about re-raising here, right after mentioning re-raising?
Or does that break the flow a bit?
Also, I think `ActiveRecord::RecordNotSaved` is still re-raised, so we can remove it here.
```suggestion
This unexpectedly breaks code that does not expect methods like `create` and
`save` to raise exceptions.

NOTE: If an exception occurs during the callback chain, Rails will re-raise
it unless it is an `ActiveRecord::Rollback` or `ActiveRecord::RecordInvalid` exception.
```